pkg/registry/core/service/storage/storage.go

```diff
@@ -248,64 +247,44 @@ func (r *REST) defaultOnReadService(service *api.Service) {
 		return
 	}
 
-	if len(service.Spec.IPFamilies) > 0 {
-		return // already defaulted
-	}
+	// If IPFamilies is set, we assume it doesn't need this at all.
```

Comment:
I think that is easier to read if we add if statements to return on what we don't want to default here:
```
if externalname {
   return
}

if len(service.Spec.IPFamilies) > 0 {
   return
}
```
it may be more verbose but at least you don't have to think what services are left from this block  :smile:

Reply:
This came up because I was talking with Andrew about maybe adding a case in here.  These returns make it hard to add to the function.  We could move this out to another function if you think it is worthwhile.  I'll add a commit.